### DISCUSSION TITLE
Using audio emulation separate

### DISCUSSION THREAD

**Discussion by emoon** [dirkwhoffmann/vAmiga#653] [General]
Hi,

I have a small project coming up that needs to *only* implement the Audio hardware (and perhaps the timers) I just wonder if it's possible to use the required bits from vAmiga (I will give include the appropriate credits of course) without needing the whole emulator or will I run into lots of troubles doing it?

**Comment by dirkwhoffmann** [comment]
>  if it's possible to use the required bits from vAmiga

I don't think it's easy. On the one hand, all components are separated nicely (I think). There is basically a class for each single hardware component. On the other hand, all components are closely tight together (just as the real hardware components are). The audio part is closely tight to the event scheduler which a part of Agnus which itself is linked to nearly everything. I think if you start separating a single part, you might end up with the whole thing. 

Nevertheless, I would be pleased to see vAmiga (or parts of it) being uses in other projects. Hence, I'd be happy if you decide to give it a try.

**Reply by emoon** [reply]
Thanks for the reply!

I may give it a go in a while and see how it goes, I will update here with the progress in that case :)
